packages/build-tools/src/steps/functions/__tests__/sendSlackMessage.test.ts

```diff
@@ -0,0 +1,227 @@
+import { errors } from '@expo/steps';
+import fetch from 'node-fetch';
+
+import { createSendSlackMessageFunction } from '../sendSlackMessage';
+import { createGlobalContextMock } from '../../../__tests__/utils/context';
+
+jest.mock('@expo/logger');
+jest.mock('node-fetch');
+
+describe(createSendSlackMessageFunction, () => {
+  let fetchMock: jest.SpyInstance;
+  let loggerInfoMock: jest.SpyInstance;
+  let loggerWarnMock: jest.SpyInstance;
+  let loggerDebugMock: jest.SpyInstance;
+  let loggerErrorMock: jest.SpyInstance;
+
+  beforeEach(() => {
+    fetchMock = jest.mocked(fetch);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  const sendSlackMessage = createSendSlackMessageFunction();
+
+  it('calls the webhook and logs the info messages when successful', async () => {
+    fetchMock.mockImplementation(() => Promise.resolve({ status: 200, ok: true, body: 'ok' }));
+    const buildStep = sendSlackMessage.createBuildStepFromFunctionCall(
+      createGlobalContextMock({}),
+      {
+        callInputs: {
+          message: 'Test message',
+        },
+        env: {
+          SLACK_HOOK_URL: 'https://slack.hook.url',
+        },
+        id: sendSlackMessage.id,
+      }
+    );
+    loggerInfoMock = jest.spyOn(buildStep.ctx.logger, 'info');
+    loggerWarnMock = jest.spyOn(buildStep.ctx.logger, 'warn');
+    await buildStep.executeAsync();
+    expect(fetchMock).toHaveBeenCalledWith('https://slack.hook.url', {
+      method: 'POST',
+      body: JSON.stringify({ text: 'Test message' }),
+      headers: { 'Content-Type': 'application/json' },
+    });
+    expect(loggerInfoMock).toHaveBeenCalledTimes(4);
+    expect(loggerInfoMock.mock.calls[0]).toEqual([
+      { marker: 'start-step' },
+      'Executing build step "Send Slack message"',
+    ]);
+    expect(loggerInfoMock.mock.calls[1]).toEqual(['Sending Slack message']);
+    expect(loggerInfoMock.mock.calls[2]).toEqual(['Slack message sent successfully']);
+    expect(loggerInfoMock.mock.calls[3]).toEqual([
+      { marker: 'end-step', result: 'success' },
+      'Finished build step "Send Slack message" successfully',
+    ]);
+    expect(loggerWarnMock).not.toHaveBeenCalled();
```

Comment:
I don't think we need to test logs to this level of detail. Same with `start-step` and `end-step` — those are not related to business logic of Slack message function so there's no reason to assert it here.